backend/de.metas.business/src/main/java/de/metas/product/model/interceptor/M_Product.java

```diff
@@ -66,4 +73,37 @@ private void createOrUpdateProductPlanningsForSelector(final @NonNull I_M_Produc
 		productPlanningSchemaBL.createOrUpdateProductPlanningsForSelector(productId, orgId, productPlanningSchemaSelector);
 	}
 
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE },
+			ifColumnsChanged = { I_M_Product.COLUMNNAME_C_UOM_ID })
+	@CalloutMethod(columnNames = I_M_Product.COLUMNNAME_C_UOM_ID)
+	public void setUOM_ID(final I_M_Product product)
+	{
+		final AdMessageKey errorMessage = checkExistingUOMConversions(product);
+
+		if (errorMessage != null)
+		{
+			throw new AdempiereException(errorMessage);
+		}
+	}
+
+	@Nullable
+	private AdMessageKey checkExistingUOMConversions(final I_M_Product product)
```

Comment:
instead, please use `IUOMConversionDAO.getProductConversions` and `UOMConversionsMap.isEmpty` to figure out it there is a conversion